### PR TITLE
‘Buying new cells’ functionality

### PR DESCRIPTION
**Pull request by jerry-sky**
I have implemented the ‘buying new map cells’ functionality on the backend. Now it would be appropriate to also implement it on the client side as well.

### CODE REVIEW

**Review by Deatomizeer — APPROVED** [review]
I'm Deatomizeer and I approve this pull request.

**Comment by jerry-sky** [comment]
Currently the user can start his city. However, they still cannot buy a new cell for their already existing city. Apart from that, lack of session retention client-side is still an on-going issue.
@KamilMatejuk 
EDIT: Actually, you can bug the client app-program by refreshing the page and _starting your city again as a non existant user of username equal to `username`_. As said above — lack of session retention is still an issue that needs to be resolved.

**Comment by KamilMatejuk** [comment]
I took a liberty to change 2 lines on the server-side, in login request returning a number of user cells, instead of boolean telling if any land is owned by a user. This was required for calculating cell cost, to keep consistent with backend.

**Comment by jerry-sky** [comment]
It does work, however I don’t understand why does the page _need_ to refresh every time I buy a new map cell.

File: client/src/app/city/city.component.ts [hunk 1]
```diff
@@ -199,13 +246,52 @@ export class CityComponent implements OnInit {
...
         // send changes to server
-        this.city.BuyBuilding(index + 1, id);
+        this.city.BuyBuilding(index + 1, id).subscribe(
+          (res) => {
+            alert(`Bought building ${id + 1}`);
+            cell.owner = this.userId;
+            cell.buildingType = id;
+            cell.buildingLvl = 0;
+            this.terrain[index] = cell;
+            window.location.reload();
```
Comment (jerry-sky) [inline client/src/app/city/city.component.ts:257]: Why does the page need to be reloaded?

**Pull request merged**